src/sinks/mod.rs

```diff
@@ -115,3 +115,201 @@ pub enum HealthcheckError {
     #[snafu(display("Unexpected status: {}", status))]
     UnexpectedStatus { status: ::http::StatusCode },
 }
+
+/*
```

Comment:
Yeah, so... this is from I originally started the PR with the intention of doing all the work, and then decided mid-way to break it down to make it easier to review.

Writing the enum is annoying as hell and involves a ton of copy and pasting, so I figured it would be fine to just keep it there knowing I'm going to uncomment it when I do the actual initial integration.

Reply:
Yep, zactly.